docs/data/material/experimental-api/css-variables/css-variables.md

```diff
@@ -0,0 +1,212 @@
+# CSS variables
+
+<p class="description">Learn about the experimental API for using CSS variables on Material UI components.</p>
+
+CSS variables provide significant improvements in developer experience related to theming and customization.
+With these variables, you can inject a theme into your app's stylesheet _at build time_ to apply the user's selected settings before the whole app is rendered.
+This solves the problem of dark-mode SSR flickering; lets you provide your users with multiple themes beyond light and dark; and offers a better debugging experience overall, among other benefits.
+
+Previously, these CSS variables were only available as an experimental API in the MUI System package.
+Now they are ready for experimental use with Material UI components.
+
+> If you want to see wider support for this API across Material UI's component library, please feel free to contribute to the ongoing development. Make sure to check the [GitHub issue](https://github.com/mui/material-ui/issues/32049) that keeps track of our progress, to see if anyone else is currently working on a component you're interested in.
+
+## Introduction
+
+The CSS variables API relies on a new experimental provider for the theme called `Experimental_CssVarsProvider` to inject styles into Material UI components.
+In addition to providing the theme in the inner React context, this new provider also generates CSS variables out of all tokens in the theme that are not functions, and makes them available in the context as well.
+
+All of these variables are accessible in an object in the theme called `vars`.
+The structure of this object is nearly identical to the theme structure, the only difference is that the values represent CSS variables.
+
+## Usage
+
+`Experimental_CssVarsProvider` is a new experimental provider that attach all generated CSS variables to the theme and put it in react context. The children under this provider will be able to read the CSS variables from the theme.
+
+```js
+import { Experimental_CssVarsProvider as CssVarsProvider } from '@mui/material/styles';
+
+function App() {
+  return <CssVarsProvider>...</CssVarsProvider>;
+}
+```
+
+### Toggle between light and dark mode
+
+`Experimental_CssVarsProvider` provides light and dark mode by default. The provider stores the user selected mode and sync with the browser local storage internally. Developers can read/update the mode via the `useColorScheme` API.
```

Comment:
I'm a little confused by this after re-reading the list of CssVarsProvider props. How does a "mode" differ from a "color scheme"? As I understand the terminology, light and dark "mode" are both types of "color schemes" (so these terms should be interchangeable) but I may not be familiar with the specific design jargon.

Reply:
This is something we had to figure out too for Joy. I'd say that my latest understanding of it is: mode is strictly either light or dark. But, each mode could have several color schemes. You can have several light mode palettes/color schemes and the same goes for dark. Does it make sense?

Reply:
That makes sense, but it seems kind of odd to me. So a dev can't create a "mode" (which could be one or more color schemes) unless it's specifically "light" or "dark"? Is a "mode" more like a "theme" then? Or it's like an organizational element that sits between the theme and the color scheme?

I think it's probably worthwhile to call out the terminology we're using here, because I'm not sure if the average docs reader would understand this distinction. Something like:

> Note: While the terms "color scheme" and "mode" may sometimes be used interchangeably, we use "mode" only when referring specifically to light and dark mode.
You may have one or more color schemes for each mode; you may also have color schemes that do not belong to any mode.

Reply:
Yeah, it's tricky 😅 And I'm pretty sure we don't have all that consistency about this topic in the docs. You'll see in our docs and elsewhere that theme and mode are also used interchangeably, both referring to dark and light modes. 

I see the theme as something that houses all the design direction—you'll find more stuff than palettes there, like spacing scale, font family, type scale, etc. Mode would be either dark or light (or auto, but that's not a third mode per se but rather the method as to how one or the other is chosen). Then, color schemes could be a synonym of palettes?! It refers to the number of schemes/palettes you have under one mode. 

Imagine you go to your computer and you choose dark mode (or appearance as Mac says). You could have several color schemes/palettes in a dark mode (fake examples: true black, grayish, cement, etc).

<img width="300" alt="Screen Shot 2022-04-04 at 15 11 43" src="https://user-images.githubusercontent.com/67129314/161606146-dd5b638a-5d4b-42c7-bad7-b617d202c881.png">

In summary, I'd say that only the last sentence is not accurate: _"you may also have color schemes that do not belong to any mode."_ Don't know how that could be. @siriwatknp your perspective here is very welcomed as we were discussing this in Joy's context as well!

Reply:
OK, so if—for example—I built an app that had four color palette options to choose from:

- light
- dark
- pink
- purple

I would have to classify "pink" and "purple" as color schemes that live within either light or dark mode?

Reply:
Yeah, you could have pink and purple on both light and dark modes, with the adequate adaptations: 

```
theme: {
   palettes: {
      light: {
        purple {...}
        pink: {...}
      }
       dark: {
        purple: {...}
        pink: {...}
     }
   }
}
```        
Does this clarifies?

Reply:
Yes, thank you! We should definitely explain this terminology in the doc here.

Reply:
Sorry to chime in later, as Danilo mentioned mode so referring to `light`/`dark` or system (for example if the users has user preference for using dark mode in the browser.

Color schemes are different palettes that can be used.

The combination of these two produces the palette that will be used in the theme :)

Reply:
Maybe an example can help. For example if there is a kids web app, you may have two different ambients for it: pink for girls and blue for boys. Both of them would support light and dark versions of them. Here is an example of how it can be implemented: https://codesandbox.io/s/bold-microservice-vgggz0?file=/src/App.js

Reply:
Cool! We could use that example. But yeah, agree with Jun that all this explanation should probably be stored elsewhere and only referenced here in this doc.